test/test_utils.py

```diff
@@ -52,3 +53,118 @@ def bootstrap():
     if not os.path.exists(test_env):
         copyfile(f"{pwd}/test/env/{test_env}", f"{test_env}")
     load_dotenv("test.env", verbose=True)
+
+
+def validate_job_state(state):
```

Comment:
I like this function. I wonder if we should use it outside of testing as well, when we return the job state.

Reply:
I think it's fine not to. If we just use it to capture all the cases where Mongo data can get mangled during testing, then using it in production would just be redundant (and would slow down things that need to be fast).